packages/editor/src/client/atoms/poll/deletePollModal.tsx

```diff
@@ -0,0 +1,86 @@
+import {ApolloError, ApolloQueryResult} from '@apollo/client'
+import React, {useEffect, useState} from 'react'
+import {useTranslation} from 'react-i18next'
+import {Button, Message, Modal, toaster} from 'rsuite'
+
+import {FullPoll, PollsQuery, useDeletePollMutation} from '../../api'
+
+interface deletePollProps {
+  poll?: FullPoll
+  setPoll(poll: FullPoll | undefined): void
+  afterDelete(): Promise<ApolloQueryResult<PollsQuery>>
+}
+
+export function DeletePollModal({poll, setPoll, afterDelete}: deletePollProps) {
+  const {t} = useTranslation()
+  const [deletePollMutation] = useDeletePollMutation()
+  const [open, setOpen] = useState<boolean>(false)
+
+  useEffect(() => {
```

Comment:
We probably should not have an useEffect here and just have open default to true.
We are not gaining much here but a double rerender.

Let's rather not render anything if poll does not exist.